modules/heat_conduction/src/kernels/AnisoHeatConduction.C

```diff
@@ -87,24 +77,26 @@ AnisoHeatConduction::computeQpResidual()
   //   }
   //   return r;
   for (unsigned i(0); i < _dim; ++i)
-  {
-    r += _grad_test[_i][_qp](i) * (*_k_i[i])[_qp] * _grad_u[_qp](i);
-  }
-  return r;
+    for (unsigned j(0); j < _dim; ++j)
+    {
+      res += _grad_test[_i][_qp](i) * (*_k(i,j))[_qp] * _grad_u[_qp](j);
+    }
```

Comment:
check out `AnisotropicDiffusion.C`

```
AnisotropicDiffusion::computeQpResidual()
{
  return (_k * _grad_u[_qp]) * _grad_test[_i][_qp];
}
```

Reply:
In the revised version of this code that I am working on `_k` is declared as `const MaterialProperty<RealTensorValue> & _k;` instead of `RealTensorValue _k;` (as in `AnisotropicDiffusion.C`). However, the multiplication does not work because of incompatible types. I am still trying to figure out a way to avoid multiple for-loops.

error: invalid operands to binary expression ('const MaterialProperty<libMesh::RealTensorValue>' (aka 'const MaterialProperty<TensorValue<double>>') and 'const libMesh::VectorValue<double>')
  return (_k * _grad_u[_qp]) * _grad_test[_i][_qp];
          ~~ ^ ~~~~~~~~~~~~

Reply:
I could not remove this error without resorting to nested for loops. So the revised version has these nested for loops.

Reply:
```
return (_k[_qp] * _grad_u[_qp]) * _grad_test[_i][_qp];
```

You forgot to index on `_qp`

Reply:
Fixed. No more nested loops :)